src/Bedrock.Framework.Experimental/Protocols/ParseResult.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using System.Runtime.CompilerServices;
+
+namespace Bedrock.Framework.Protocols.Http.Http1
+{
+    public struct ParseResult<T>
+    {
+        private T _value;
+        private Exception _error;
```

Comment:
Do we want to use Exceptions here? Or something else?

Reply:
Feels wrong to have an exception that isn't thown.